src/services/composite-keys-manager.js

```diff
@@ -1,54 +1,83 @@
 import _ from 'lodash';
 import Operators from '../utils/operators';
 
-function CompositeKeysManager(model, schema, record) {
-  const GLUE = '|';
-
-  this.getPrimaryKeyValues = function getPrimaryKeyValues(recordId) {
-    const primaryKeyValues = recordId.split(GLUE);
-
-    // NOTICE: Prevent liana to crash when a composite primary keys is null,
-    //         this behaviour should be avoid instead of fixed.
-    primaryKeyValues.forEach((key, index) => {
-      if (key === 'null') { primaryKeyValues[index] = null; }
-    });
-
-    return primaryKeyValues;
-  };
-
-  this.getRecordsConditions = function getRecordsConditions(recordsIds, options) {
-    const { OR } = Operators.getInstance(options);
-    return { [OR]: recordsIds.map((recordId) => this.getRecordConditions(recordId)) };
-  };
-
-  this.getRecordConditions = function getRecordConditions(recordId) {
-    const where = {};
-    const primaryKeyValues = this.getPrimaryKeyValues(recordId);
-    if (primaryKeyValues.length === _.keys(model.primaryKeys).length) {
-      _.keys(model.primaryKeys).forEach((primaryKey, index) => {
-        where[primaryKey] = primaryKeyValues[index];
+/**
+ * This helper class allows abstracting away the complexity
+ * of using collection which have composite primary keys.
+ */
+class CompositeKeysManager {
```

Comment:
Why not using the dependency injector like the new services 🤔  ?

Reply:
There is no dependency injection in this project at all.
Maybe because forest-express and forest-express-sequelize depend on each other and it was difficult to set up?